### PR TITLE
 plan: fix a bug in index join

### PR DESCRIPTION
**Pull request by winoros**
## What have you changed? (mandatory)

wrongly usage of golang's `copy`

## What is the type of the changes? (mandatory)

- Bug fix

## How has this PR been tested? (mandatory)

newly added explain test.

### CODE REVIEW

**Comment by XuHuaiyu** [comment]
ci failed @winoros

File: expression/explain.go [hunk 1]
```diff
@@ -60,12 +61,26 @@ func (expr *Constant) format(dt types.Datum) string {
 }
 
 // ExplainExpressionList generates explain information for a list of expressions.
-func ExplainExpressionList(exprs []Expression) []byte {
+func ExplainExpressionList(exprs []Expression, needSort bool) []byte {
```
Comment (XuHuaiyu) [inline expression/explain.go]: add  a comment for needSort

**Comment by winoros** [comment]
@XuHuaiyu PTAL

File: expression/explain.go [hunk 2]
```diff
@@ -60,12 +61,28 @@ func (expr *Constant) format(dt types.Datum) string {
 }
 
 // ExplainExpressionList generates explain information for a list of expressions.
-func ExplainExpressionList(exprs []Expression) []byte {
+// needSort will be true if the exprs should keep its order. In some scenarios, the expr's order may not be valid
```
Comment (zz-jason) [inline expression/explain.go]: why the order of the `exprs` may not be valid?

Comment (zz-jason) [reply expression/explain.go]: Uh, I think `stable` is a better word. By the way, this function only used when we `explain` a sql, and the performance doesn't matter, I think we can force to sort the `exprs` to simplify the function signature, and this function can be renamed to `SortAndExplainExpressionList`.

**Pull request merged**